src/utils/collective_evaluation.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright © 2023 Niklas Siemer
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This module contains functions to collectively evaluate parameters
+//! instead of copying the same lines of code for error handling in each function.
+
+use crate::error::MathError;
+
+/// Evaluates whether `row0` and `row1` is smaller than `self_num_rows` and `other_num_rows`.
+/// Additionally, whether `self_num_cols` equals `other_num_cols`.
+///
+/// Parameters:
+/// - `callee`: specifies the name of the calling function
+/// - `row0`: specifies a value that should be smaller than `self_num_rows`
+/// - `row1`: specifies a value that should be smaller than `other_num_rows`
+/// - `self_num_rows`: is evaluated against `row0`
+/// - `other_num_rows`: is evaluated against `row1`
+/// - `self_num_cols`: is used to check whether it equals `other_num_cols`
+/// - `other_num_cols`: is evaluated against `self_num_cols`
+///
+/// Returns an empty `Ok` if `row0 < self_num_rows`, `row1 < other_num_rows`, and `self_num_cols == other_num_cols`.
+/// Otherwise, a [`MathError`] is returned.
+///
```

Comment:
This only describes funtionality, could you describe some of its purpose as well